### PR TITLE
Possibile fix for  variable not set in page

### PR DESCRIPTION
**Pull request by ChristianGiupponi**
I have added a new method in the BasePublicController to allow all the PublicController to add alternate urls.  
By default the $alternateUrls array is empty and is shared in the view with the share method.  

Don't know if is the best way, please check it

Refer: https://github.com/AsgardCms/Platform/issues/631

### CODE REVIEW

File: Modules/Page/Http/Controllers/PublicController.php [hunk 1]
```diff
@@ -96,12 +98,13 @@ private function getTemplateForPage($page)
...
     /**
-     * Throw a 404 error page if the given page is not found or draft
+     * Throw a 404 error page if the given page is not found
+     *
      * @param $page
      */
     private function throw404IfNotFound($page)
     {
-        if (null === $page || $page->status === $this->disabledPage) {
+        if (is_null($page)) {
```
Comment (mikemand) [inline Modules/Page/Http/Controllers/PublicController.php]: Why did you remove the check for a disabled page? If a page's status is set to `false`, we don't want someone to be able to access it at all.

File: Modules/Core/Http/Controllers/BasePublicController.php [hunk 2]
```diff
@@ -14,9 +18,67 @@
...
+
+        $this->addAssets();
+        $this->requireDefaultAssets();
+        view()->share('alternate', $this->alternateUrls);
+    }
+
+    /**
+     * Add the assets from the config file on the asset manager.
+     */
+    private function addAssets()
```
Comment (mikemand) [inline Modules/Core/Http/Controllers/BasePublicController.php]: Why all this extra assets stuff? Pretty sure it isn't used in the front end...

File: Modules/Page/Http/Controllers/PublicController.php (hunk 1 above)
Comment (ChristianGiupponi) [reply Modules/Page/Http/Controllers/PublicController.php]: Rally don't no, It was mu default controller, I didn't removed any of them

File: Modules/Core/Http/Controllers/BasePublicController.php (hunk 2 above)
Comment (ChristianGiupponi) [reply Modules/Core/Http/Controllers/BasePublicController.php]: Same as above, maybe I have some old controller, tomorrow I will remove this request and try to understand whats wring with my branch.
Beside this, have a look to the solution if it can work
